drivers/ieee802154/ieee802154_cc13xx_cc26xx.c

```diff
@@ -225,9 +265,12 @@ static int ieee802154_cc13xx_cc26xx_tx(const struct device *dev,
 	__ASSERT_NO_MSG(k_sem_count_get(&drv_data->fg_done) == 0);
 
 	do {
```

Comment:
@bwitherspoon @vanti - I wonder if it would be possible to convert the TX ISR to a SWI callback as well, sort of like what I did for the RX ISR. The RF core supports a number of configurable conditions.. I wonder if it can count down too..

Edit:  Aha!
https://www.ti.com/lit/pdf/swcu185
25.3.3.1.11 CMD_COUNT: Counter Command

Reply:
> 25.3.3.1.11 CMD_COUNT: Counter Command

Actually
25.3.3.1.13 CMD_COUNT_BRANCH
might be more appropriate. Also
25.3.3.1.14 CMD_PATTERN_CHECK.

It looks as though the `RF_runCmd()` call could be used to run a chain of commands of the form:

CSMA -> IEEE_TX -> IEEE_RX_ACK -> PATTERN_CHECK -> CMD_COUNT_BRANCH
or
CSMA -> IEEE_TX -> PATTERN_CHECK -> CMD_COUNT_BRANCH

based on whether an ACK was requested. The PATTERN_CHECK pattern would check to see if IEEE_TX had been successful.

The pNextOp for PATTERN_CHECK would be NULL if the IEEE_TX were successful. Otherwise, we would get to the COUNT_BRANCH command.

The pNextOp for COUNT_BRANCH would be IEEE_TX again if the count were non-zero.

Should be interesting to see if there is a performance improvement.

Reply:
The radio command graph I was thinking for this would be something along these lines:
```graphviz
digraph G {

  start -> CSMA;
  CSMA -> "PATTERN_CHECK\nCSMA OK?";
  "PATTERN_CHECK\nCSMA OK?" -> CSMA [label = "no"];
  "PATTERN_CHECK\nCSMA OK?" -> IEEE_TX [label = "yes"];
  IEEE_TX -> "PATTERN_CHECK\nIEEE_TX ok?";
  "PATTERN_CHECK\nIEEE_TX ok?" -> CSMA [label = "no"];
  "PATTERN_CHECK\nIEEE_TX ok?" -> IEEE_RX_ACK [label = "yes"];
  IEEE_RX_ACK -> "PATTERN_CHECK\nIEEE_RX_ACK ok?";
  "PATTERN_CHECK\nIEEE_RX_ACK ok?" -> "COUNT_BRANCH\nRADIO_TX_RETRIES" [label = "no"];
  "PATTERN_CHECK\nIEEE_RX_ACK ok?" -> end [label = "yes"];
  "COUNT_BRANCH\nRADIO_TX_RETRIES" -> CSMA [label = "> 0"];
  "COUNT_BRANCH\nRADIO_TX_RETRIES" -> end [label = "== 0"];
  
  CSMA[shape="rectangle"];
  "PATTERN_CHECK\nCSMA OK?"[shape="diamond"];
  IEEE_TX[shape="rectangle"];
  "PATTERN_CHECK\nIEEE_TX ok?"[shape="diamond"];
  IEEE_RX_ACK[shape="rectangle"];
  "PATTERN_CHECK\nIEEE_RX_ACK ok?"[shape="diamond"];
  "COUNT_BRANCH\nRADIO_TX_RETRIES"[shape="diamond"];
}
```
https://dreampuf.github.io/GraphvizOnline

Reply:
The radio command graph should look something like this:

```dot
digraph G {

  start -> CSMA;
  CSMA -> "PATTERN_CHECK\nCSMA OK?";
  "PATTERN_CHECK\nCSMA OK?" -> "COUNT_BRANCH\nRADIO_TX_RETRIES" [label = "no"];
  "PATTERN_CHECK\nCSMA OK?" -> IEEE_TX [label = "yes"];
  IEEE_TX -> "PATTERN_CHECK\nIEEE_TX ok?";
  "PATTERN_CHECK\nIEEE_TX ok?" -> "COUNT_BRANCH\nRADIO_TX_RETRIES" [label = "no"];
  "PATTERN_CHECK\nIEEE_TX ok?" -> IEEE_RX_ACK [label = "yes"];
  IEEE_RX_ACK -> "PATTERN_CHECK\nIEEE_RX_ACK ok?";
  "PATTERN_CHECK\nIEEE_RX_ACK ok?" -> "COUNT_BRANCH\nRADIO_TX_RETRIES" [label = "no"];
  "PATTERN_CHECK\nIEEE_RX_ACK ok?" -> end [label = "yes"];
  "COUNT_BRANCH\nRADIO_TX_RETRIES" -> CSMA [label = "> 0"];
  "COUNT_BRANCH\nRADIO_TX_RETRIES" -> end [label = "== 0"];
  
  CSMA[shape="rectangle"];
  "PATTERN_CHECK\nCSMA OK?"[shape="diamond"];
  IEEE_TX[shape="rectangle"];
  "PATTERN_CHECK\nIEEE_TX ok?"[shape="diamond"];
  IEEE_RX_ACK[shape="rectangle"];
  "PATTERN_CHECK\nIEEE_RX_ACK ok?"[shape="diamond"];
  "COUNT_BRANCH\nRADIO_TX_RETRIES"[shape="diamond"];
}
```
https://dreampuf.github.io/GraphvizOnline